libmscore/album.cpp

```diff
@@ -0,0 +1,1489 @@
+            composers.push_back(composer);
+        }
+    }
+
+    return composers;
+}
+
+//---------------------------------------------------------
+//   lyricists
+//---------------------------------------------------------
+
+QStringList Album::lyricists() const
+{
+    QStringList lyricists;
+
+    for (auto& item : m_albumItems) {
+        if (!item->checkReadiness()) {
+            continue;
+        }
+        QString lyricist = item->score()->lyricist();
+        if (!lyricists.contains(lyricist) && !lyricist.isEmpty()) {
+            lyricists.push_back(lyricist);
+        }
+    }
+
+    return lyricists;
+}
+
+//---------------------------------------------------------
+//   scoreTitles
+//---------------------------------------------------------
+
+QStringList Album::scoreTitles() const
+{
+    QStringList scoreTitles;
+
+    for (auto& item : m_albumItems) {
+        if (!item->checkReadiness()) {
+            continue;
+        }
+        if (!item->score()->textMovement()) {
+            QString title = item->score()->realTitle();
+            title = title.isEmpty() ? item->score()->title() : title;
+            scoreTitles.push_back(title);
+        }
+    }
+
+    return scoreTitles;
+}
+
+//---------------------------------------------------------
+//   checkPartCompatibility
+///     Check if the scores in the Album have compatible parts
+//---------------------------------------------------------
+
+bool Album::checkPartCompatibility() const
+{
+    if (!m_albumItems.size()) {
+        return true;
+    }
+
+    MasterScore* firstMovement = m_albumItems.at(0)->score();
+    int partCount = firstMovement->parts().size();
+    // check number of parts
+    for (auto& ms : albumScores()) {
+        if (partCount < ms->parts().size()) {
+            return false;
+        }
+    }
+    // check part names
+    for (int i = 0; i < partCount; i++) {
+        for (auto& ms : albumScores()) {
+            if (ms->parts().at(i)->partName().compare(firstMovement->parts().at(i)->partName(),
+                                                      Qt::CaseSensitivity::CaseInsensitive)) {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+//---------------------------------------------------------
+//   checkPartCompatibility
+///     Check if the given score breaks part compatibility
+//---------------------------------------------------------
+
+bool Album::checkPartCompatibility(MasterScore* score)
+{
+    if (!m_albumItems.size()) {
+        return true;
+    }
+
+    // check if the new score breaks compatibility
+    MasterScore* firstMovement = m_albumItems.at(0)->score();
+    int partCount = firstMovement->parts().size();
+    // check number of parts
+    if (partCount < score->parts().size()) {
+        return false;
+    }
+    // check part names
+    for (int i = 0; i < partCount; i++) {
+        if (score->parts().at(i)->partName().compare(firstMovement->parts().at(i)->partName(),
```

Comment:
This of course assumes that the user hasn't changed the part's name, which may not be a very safe assumption to make...

Reply:
I agree, I want to change this. I just could not find (even though I am sure it exists) the instrument id. I think I just found it. The correct way to do this would be to get the instrumentList of the part and compare instrument ids right?

Reply:
I'd imagine so, yes. Instrument ids should be constant, whereas user-editable strings... not so much ;)